### ISSUE TITLE
config: add allowed-hosts to whitelist secret-exfiltration destinations

### ISSUE THREAD

**Issue by HikaruEgashira** [ultra-supara/sisakulint#473]
## Summary

Add an `allowed-hosts` configuration option (in `.github/action.yaml` and/or per-workflow inline) to whitelist trusted destinations for the `secret-exfiltration` rule. This lets users keep `fail-on: high` (default) while permitting known-safe outbound calls like a vendor's own API.

## Background

`secret-exfiltration` fires `critical` whenever a `${{ secrets.X }}`-derived env var is used with `curl`/`wget`/etc. against any host. This is correct in general — but produces unavoidable false positives when the destination **is** the legitimate API for that secret. For example, a workflow that needs to call a vendor's REST API with the matching vendor API token; that is the only thing the token does.

Today users have to either:

1. Refactor the workflow to remove `curl` entirely (e.g. push everything through a provider's terraform/`*-action` even when the operation is a one-off import), or
2. Drop `-ignore secret-exfiltration` at the wrapper layer, which is far too broad — it disables the rule for any other host in the workflow as well.

Neither matches the intent of the rule.

### Config file (`.github/action.yaml`)

```yaml
secret-exfiltration:
  allowed-hosts:
    # Exact match
    - api.example.com
    # Suffix wildcard
    - "*.example.com"
```

A finding is suppressed if the destination URL hostname matches any entry. Match is on the literal URL substring that follows `https://` / `http://`, scoped to a host token. No regex — keep the surface small and review-friendly.

### Per-workflow override

For repos that want the allowlist scoped to a single workflow (e.g. terraform-only file), allow the same key at workflow level:

```yaml
# .github/workflows/deploy.yml
# sisakulint:secret-exfiltration.allowed-hosts: api.example.com
```

(Or a top-level YAML key like `x-sisakulint:` — bikeshed welcome.)

### Behaviour

- A finding is still reported as a warning in `verbose` mode so it's visible, just not gated.
- Auto-fix is unaffected (this rule has none).
- An entry that doesn't match any actual `curl`/`wget` site in the workflow is itself a warning (dead allow), so the list doesn't rot.

## Why this is better than `-ignore secret-exfiltration`

- `-ignore secret-exfiltration` disables the rule for the whole workflow including future, unrelated `curl` lines that *would* exfiltrate. The allowed-hosts list is positive (allow only these), not negative.
- It's reviewable in PR diffs — a new entry needs justification.
- It's portable: same expression for shared and per-workflow scope.

## Concrete motivating diff

Before (no way to express "this curl to api.example.com is fine"):

```yaml
curl -s "https://api.example.com/v1/resources" \
  -H "Authorization: Bearer $VENDOR_API_TOKEN"
# → secret-exfiltration (critical) — fail-on: high → CI red
```

After:

```yaml
# .github/action.yaml
secret-exfiltration:
  allowed-hosts:
    - api.example.com

# unchanged workflow — passes
```

## Open questions

- Should `allowed-hosts` apply to `secret-in-log` too (when the shellvar origin is `curl-to-allowed-host`)? Probably yes — derived values from a trusted call are no longer "secret-tainted."
- Scope: per-rule (`secret-exfiltration.allowed-hosts`) vs. a shared `trusted-hosts:` top-level list reused by multiple rules. I'd lean shared.
